### PR TITLE
Add cpg-id-checker

### PR DESCRIPTION
**Pull request by illusional**
I've tagged a 0.1.2 to test with this, and have applied to TOB repo, like:

```yaml
  - repo: https://github.com/populationgenomics/pre-commits
    rev: "v0.1.2"
    hooks:
      - id: cpg-id-checker
        args: ["--extra-pattern", 'TOB\d+']
```

This does in fact actually work 🎉 

I've added a README and a test workflow.

### CODE REVIEW

**Review by MattWellie — APPROVED** [review]
Looks awesome!

**Comment by illusional** [comment]
Hey @MattWellie, I've added a test workflow and a README, so probably worth getting another review :)

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          files: ./coverage.xml
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+
+      - name: Fail if tests are not passing
+        if: ${{ steps.runtests.outputs.rc != 0 }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.setFailed('Unit tests failed with rc = ${{ steps.runtests.outputs.rc }}')
```
Comment (MattWellie) [inline .github/workflows/test.yaml:47]: that's cool

File: README.md [hunk 2]
```diff
@@ -1,6 +1,39 @@
-# CPG Python Template
+# Pre-commits
 
-Template repository based on the [CPG team-docs](https://github.com/populationgenomics/team-docs/blob/main/new_repository.md) content.
+This repository contains
```
Comment (MattWellie) [inline README.md]: 🤔

**Review by MattWellie — APPROVED** [review]
I still like it, but an empty chunk in the README.md

**Comment by MattWellie** [comment]
@illusional I've just run this on the AIP repo, and a couple of things popped up 

- it tries to read all compressed files (.gz, .bgz) as plain text, so runs into a decoding error during iteration
- it doesn't greenlist any file extensions, so it also barfed trying to scan PNG files etc.

I'd like to shove a few escapes/alternate parsers in here for those situations, not sure if you want to take the lead and own this, or happy for me to have a poke (will make a separate branch if so, to merge in here)

**Review by MattWellie — APPROVED** [review]
LGTMatt

**Pull request merged**